ui/pages/confirmations/components/confirm/info/hooks/useUniswapShieldInfo.ts

```diff
@@ -0,0 +1,222 @@
+import { Hex } from '@metamask/utils';
+import { Interface, TransactionDescription } from '@ethersproject/abi';
+import { TransactionMeta } from '@metamask/transaction-controller';
+import { getNativeTokenAddress } from '@metamask/assets-controllers';
+import { useEffect } from 'react';
+
+import { Numeric } from '../../../../../../../shared/modules/Numeric';
+import { fetchQuotes } from '../../../../../../store/actions';
+import { useConfirmContext } from '../../../../context/confirm';
+import { useTransactionEventFragment } from '../../../../hooks/useTransactionEventFragment';
+
+const ABI = [
+  {
+    constant: true,
+    inputs: [
+      {
+        name: 'commands',
+        type: 'bytes',
+      },
+      {
+        name: 'inputs',
+        type: 'bytes[]',
+      },
+      {
+        name: 'deadline',
+        type: 'uint256',
+      },
+    ],
+    name: 'execute',
+    type: 'function',
+  },
+  {
+    constant: true,
+    inputs: [
+      {
+        name: 'commands',
+        type: 'bytes',
+      },
+      {
+        name: 'inputs',
+        type: 'bytes[]',
+      },
+    ],
+    name: 'execute',
+    type: 'function',
+  },
+];
+
+const getWordsFromInput = (input: string) => {
+  return input.slice(2).match(/.{1,64}/g) as string[];
+};
+
+function wordToAddress(word: string) {
+  return '0x' + word.slice(24);
+}
+
+function wordToAmount(word: string) {
+  const amount = word.replace(/^0+/, '');
+  return '0x' + amount;
+}
+
+export function useUniswapShieldInfo() {
+  const { currentConfirmation } = useConfirmContext<TransactionMeta>();
+  const {
+    chainId,
+    id: transactionId,
+    simulationData,
+    txParams,
+  } = currentConfirmation ?? {
+    txParams: {},
+  };
+  const { data, gas, value: amount } = txParams ?? {};
+  const contractInterface = new Interface(ABI);
+  const { updateTransactionEventFragment } = useTransactionEventFragment();
+
+  useEffect(() => {
+    updateTransactionEventFragment(
+      {
+        properties: {
+          uniswap_shield: 'loading',
+        },
+      },
+      transactionId,
+    );
+
+    let parsedTransactionData: TransactionDescription;
+
+    try {
+      parsedTransactionData = contractInterface.parseTransaction({
+        data: data as Hex,
+      });
+    } catch (error) {
+      return;
+    }
+
+    const commands = parsedTransactionData.args.commands as string;
```

Comment:
However we extract the necessary data, should all the Uniswap logic remain encapsulated in `app/scripts/lib/transaction/decode/uniswap.ts`?

Maybe with a new `getUniswapSwapData` function?

Reply:
This is not standard parsing of abi data and will not be re-used I think. Additionally putting data in `uniswap.ts` will then need for a controller action to be called from the client.